src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigsCollection.java

```diff
@@ -85,50 +49,52 @@ public String toPackedFasta(){
      * @param packedFastaLine
      */
     @VisibleForTesting
-    protected static ContigsCollection fromPackedFasta(final String packedFastaLine) {
+    protected static ContigsCollection fromPackedFasta(final long assemblyID, final String packedFastaLine) {
         final List<String> fileContents = Arrays.asList(packedFastaLine.split("\\|"));
         if (fileContents.size() % 2 != 0) {
             throw new GATKException("Odd number of lines in breakpoint fasta" + packedFastaLine);
         }
-        return new ContigsCollection(fileContents);
+        return new ContigsCollection(assemblyID, fileContents);
     }
 
-    public List<String> toListOfStrings(){
-        if(null==contents){
+    /**
+     * @return {@code null} when {@link #assemblyContents} is,
+     *          otherwise a list of string representation of contig id, and contig sequence
+     */
+    List<String> toListOfStrings(){
+        if(null==assemblyContents){
             return null;
         }
-        final List<String> res = new ArrayList<>();
-        for(final Tuple2<ContigID, ContigSequence> contig : contents){
-            res.add(contig._1().toString());
-            res.add(contig._2().toString());
-        }
-        return res;
+        return assemblyContents.stream().flatMap(contig -> Stream.of(contig.contigID, contig.seq)).collect(Collectors.toList());
     }
 
-    public ContigsCollection(final List<String> fileContents){
+    /**
+     * Ctor to be used for SGA output
+     */
+    ContigsCollection(final long assemblyID, final List<String> sgaFASTAFileContents){
```

Comment:
Yes, admittedly the documentation is bad. I was thinking too much about the different outputs between SGA and fermi-lite.